flink-runtime/src/main/java/org/apache/flink/runtime/metrics/dump/MetricDump.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.runtime.metrics.dump;
+
+/**
+ * A container for a dumped metric that contains the scope, name and value(s) of the metric.
+ */
+public abstract class MetricDump {
+	/** Categories to be returned by {@link MetricDump#getCategory()} to avoid instanceof checks. */
+	public static final byte METRIC_CATEGORY_COUNTER = 0;
+	public static final byte METRIC_CATEGORY_GAUGE = 1;
+	public static final byte METRIC_CATEGORY_HISTOGRAM = 2;
+
+	/** The scope information for the stored metric. */
+	public final QueryScopeInfo scopeInfo;
+	/** The name of the stored metric. */
+	public final String name;
+
+	private MetricDump(QueryScopeInfo scopeInfo, String name) {
+		this.scopeInfo = scopeInfo;
+		this.name = name;
+	}
+
+	/**
+	 * Returns the category for this MetricDump.
+	 *
+	 * @return category
+	 */
+	public abstract byte getCategory();
```

Comment:
I think we don't need the explicit category information, because it is already encoded in the sub-types of `MetricDump`.